app/src/main/java/com/github/featuredetectandroid/ui/Menu.kt

```diff
@@ -0,0 +1,84 @@
+package com.github.featuredetectandroid.ui
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.github.featuredetectandroid.utils.KeypointDetectionAlgorithm
+
+@Composable
+fun Menu(currentAlgorithm: String, onAlgorithmSelected: (String) -> Unit) {
+    val radioOptions = KeypointDetectionAlgorithm.names
+    var selectedAlgorithm by remember { mutableStateOf(currentAlgorithm) }
```

Comment:
I think, it would be better to put this outside of this composable -- this better corresponds to the state hoisting pattern

Reply:
fixed